From c89ce946e5328ca8a7df923d421e904bb6bfe9b6 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Tue, 7 Jul 2015 18:28:46 +0200
Subject: [PATCH 63/76] safe_browsing: disable reporting of safebrowsing
 override

Disables reporting of the safebrowsing override, i.e. the report sent
if a user decides to visit a page that was flagged as "insecure".
This prevents trk:148 (phishing) and trk:149 (malware).
Also disables Safe Browsing incident reporting and its opt-in UI.
--- a/chrome/browser/safe_browsing/chrome_safe_browsing_blocking_page_factory.cc
+++ b/chrome/browser/safe_browsing/chrome_safe_browsing_blocking_page_factory.cc
@@ -63,8 +63,7 @@ ChromeSafeBrowsingBlockingPageFactory::C
       Profile::FromBrowserContext(web_contents->GetBrowserContext());
   // Create appropriate display options for this blocking page.
   PrefService* prefs = profile->GetPrefs();
-  bool is_extended_reporting_opt_in_allowed =
-      IsExtendedReportingOptInAllowed(*prefs);
+  bool is_extended_reporting_opt_in_allowed = false;
   bool is_proceed_anyway_disabled = IsSafeBrowsingProceedAnywayDisabled(*prefs);
 
   // Determine if any prefs need to be updated prior to showing the security
--- a/chrome/browser/safe_browsing/incident_reporting/incident_report_uploader_impl.cc
+++ b/chrome/browser/safe_browsing/incident_reporting/incident_report_uploader_impl.cc
@@ -29,7 +29,7 @@ namespace safe_browsing {
 namespace {
 
 const char kSbIncidentReportUrl[] =
-    "https://sb-ssl.google.com/safebrowsing/clientreport/incident";
+    "trk:268:https://sb-ssl.google.com/safebrowsing/clientreport/incident";
 
 constexpr net::NetworkTrafficAnnotationTag
     kSafeBrowsingIncidentTrafficAnnotation =
--- a/chrome/browser/safe_browsing/incident_reporting/incident_reporting_service.cc
+++ b/chrome/browser/safe_browsing/incident_reporting/incident_reporting_service.cc
@@ -301,12 +301,8 @@ IncidentReportingService::UploadContext:
 IncidentReportingService::UploadContext::~UploadContext() = default;
 
 // static
-bool IncidentReportingService::IsEnabledForProfile(Profile* profile) {
-  if (profile->IsOffTheRecord())
-    return false;
-  if (!IsSafeBrowsingEnabled(*profile->GetPrefs()))
-    return false;
-  return IsExtendedReportingEnabled(*profile->GetPrefs());
+bool IncidentReportingService::IsEnabledForProfile(Profile*) {
+  return false;
 }
 
 IncidentReportingService::IncidentReportingService(
--- a/chrome/browser/safe_browsing/safe_browsing_service.cc
+++ b/chrome/browser/safe_browsing/safe_browsing_service.cc
@@ -500,11 +500,7 @@ SafeBrowsingUIManager* SafeBrowsingServi
       chrome::ChromeUINewTabURLAsGURL());
 }
 
-void SafeBrowsingServiceImpl::RegisterAllDelayedAnalysis() {
-#if BUILDFLAG(FULL_SAFE_BROWSING)
-  RegisterBinaryIntegrityAnalysis();
-#endif
-}
+void SafeBrowsingServiceImpl::RegisterAllDelayedAnalysis() {}
 
 V4ProtocolConfig SafeBrowsingServiceImpl::GetV4ProtocolConfig() const {
   return safe_browsing::GetV4ProtocolConfig();
--- a/components/safe_browsing/core/browser/client_side_detection_service_base.cc
+++ b/components/safe_browsing/core/browser/client_side_detection_service_base.cc
@@ -461,113 +461,11 @@ void ClientSideDetectionServiceBase::Sen
     ClientReportPhishingRequestCallback callback,
     const std::string& access_token) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
-  if (!IsEnabled()) {
     if (!callback.is_null()) {
       base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
           FROM_HERE, base::BindOnce(std::move(callback), GURL(verdict->url()),
                                     false, std::nullopt, std::nullopt));
     }
-    return;
-  }
-
-  // If the report was not requested by the user, record that we made a request.
-  // Logged before the request is made to ensure it gets recorded. If this
-  // returns false due to being at ping cap or prefs are null, abandon the
-  // request.
-  if (verdict->client_side_detection_type() !=
-          ClientSideDetectionType::USER_REPORT &&
-      !AddPhishingReport(base::Time::Now())) {
-    if (!callback.is_null()) {
-      base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
-          FROM_HERE, base::BindOnce(std::move(callback), GURL(verdict->url()),
-                                    false, std::nullopt, std::nullopt));
-    }
-    return;
-  }
-
-  std::string request_data;
-  verdict->SerializeToString(&request_data);
-
-  net::NetworkTrafficAnnotationTag traffic_annotation =
-      net::DefineNetworkTrafficAnnotation(
-          "safe_browsing_client_side_phishing_detector", R"(
-          semantics {
-            sender: "Safe Browsing Client-Side Phishing Detector"
-            description:
-              "If the client-side phishing detector determines that the "
-              "current page contents are similar to phishing pages, it will "
-              "send a request to Safe Browsing to ask for a final verdict. If "
-              "Safe Browsing agrees the page is dangerous, Chrome will show a "
-              "full-page interstitial warning."
-            trigger:
-              "Whenever the client-side detector machine learning model "
-              "computes a phishy-ness score above a threshold, after page-load."
-            internal {
-              contacts {
-                email: "chrome-counter-abuse-alerts@google.com"
-              }
-            }
-            user_data {
-              type: ACCESS_TOKEN
-              type: SENSITIVE_URL
-              type: WEB_CONTENT
-            }
-            data:
-              "Top-level page URL without CGI parameters, boolean and double "
-              "features extracted from DOM, such as the number of resources "
-              "loaded in the page, if certain likely phishing and social "
-              "engineering terms found on the page, etc."
-            destination: GOOGLE_OWNED_SERVICE
-            last_reviewed: "2026-06-17"
-          }
-          policy {
-            cookies_allowed: YES
-            cookies_store: "Safe browsing cookie store"
-            setting:
-              "Users can enable or disable this feature by toggling 'Protect "
-              "you and your device from dangerous sites' in Chrome settings "
-              "under Privacy. This feature is enabled by default."
-            chrome_policy {
-              SafeBrowsingProtectionLevel {
-                policy_options {mode: MANDATORY}
-                SafeBrowsingProtectionLevel: 0
-              }
-            }
-            chrome_policy {
-              SafeBrowsingEnabled {
-                policy_options {mode: MANDATORY}
-                SafeBrowsingEnabled: false
-              }
-            }
-            deprecated_policies: "SafeBrowsingEnabled"
-          })");
-
-  auto resource_request = std::make_unique<network::ResourceRequest>();
-  if (!access_token.empty()) {
-    SetAccessToken(resource_request.get(), access_token);
-  }
-
-  resource_request->url = GetClientReportUrl(kClientReportPhishingUrl);
-  resource_request->method = "POST";
-  resource_request->load_flags = net::LOAD_DISABLE_CACHE;
-
-  auto loader = network::SimpleURLLoader::Create(std::move(resource_request),
-                                                 traffic_annotation);
-  loader->AttachStringForUpload(request_data, "application/octet-stream");
-  loader->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
-      url_loader_factory_.get(),
-      base::BindOnce(&ClientSideDetectionServiceBase::OnURLLoaderComplete,
-                     base::Unretained(this), loader.get(), base::Time::Now()));
-
-  // Remember which callback and URL correspond to the current fetcher object.
-  auto info = std::make_unique<ClientPhishingReportInfo>();
-  auto* loader_ptr = loader.get();
-  info->loader = std::move(loader);
-  info->callback = std::move(callback);
-  info->phishing_url = GURL(verdict->url());
-  client_phishing_reports_[loader_ptr] = std::move(info);
-
-  DidSendClientReportPhishingRequest(std::move(verdict), access_token);
 }
 
 CSDModelType ClientSideDetectionServiceBase::GetModelType() const {
--- a/components/security_interstitials/core/safe_browsing_loud_error_ui.cc
+++ b/components/security_interstitials/core/safe_browsing_loud_error_ui.cc
@@ -27,11 +27,11 @@ namespace {
 // For malware interstitial pages, we link the problematic URL to Google's
 // diagnostic page.
 const char kSbDiagnosticUrl[] =
-    "https://transparencyreport.google.com/safe-browsing/search?url=%s";
+    "trk:227:https://transparencyreport.google.com/safe-browsing/search?url=%s";
 
 // Constants for the V4 phishing string upgrades.
 const char kReportPhishingErrorUrl[] =
-    "https://safebrowsing.google.com/safebrowsing/report_error/?url=%s";
+    "trk:228:https://safebrowsing.google.com/safebrowsing/report_error/?url=%s";
 
 void RecordExtendedReportingPrefChanged(bool report) {
   UMA_HISTOGRAM_BOOLEAN("SafeBrowsing.Pref.Extended.SecurityInterstitial",
